tests/flow/test_concurrent_query.py

```diff
@@ -246,96 +224,95 @@ def test_05_concurrent_read_delete(self):
     def test_06_concurrent_write_delete(self):
         # Test setup - validate that graph exists and possible results are None
         graphs[0].query("MATCH (n) RETURN n")
-        assertions[0] = None
-        exceptions[0] = None
 
+        pool = Pool(nodes=1)
         redis_con = self.env.getConnection()
         heavy_write_query = """UNWIND(range(0,999999)) as x CREATE(n)"""
-        writer = threading.Thread(target=thread_run_query, args=(graphs[0], heavy_write_query, 0))
-        writer.setDaemon(True)
-        writer.start()
+        writer = pool.apipe(thread_run_query, graphs[0], heavy_write_query, 0)
         redis_con.delete(GRAPH_ID)
-        writer.join()
+        writer.wait()
         possible_exceptions = ["Encountered different graph value when opened key " + GRAPH_ID,
                                "Encountered an empty key when opened key " + GRAPH_ID]
-        if exceptions[0] is not None:
-            self.env.assertContains(exceptions[0], possible_exceptions)
+        result = writer.get()
+        if isinstance(result, str):
+            self.env.assertContains(result, possible_exceptions)
         else:
-            self.env.assertEquals(1000000, assertions[0].nodes_created)       
+            self.env.assertEquals(1000000, result.nodes_created)
     
     def test_07_concurrent_write_rename(self):
         # Test setup - validate that graph exists and possible results are None
         graphs[0].query("MATCH (n) RETURN n")
-        assertions[0] = None
-        exceptions[0] = None
 
+        pool = Pool(nodes=1)
         redis_con = self.env.getConnection()
         new_graph = GRAPH_ID + "2"
         # Create new empty graph with id GRAPH_ID + "2"
         redis_con.execute_command("GRAPH.QUERY",new_graph, """MATCH (n) return n""", "--compact")
         heavy_write_query = """UNWIND(range(0,999999)) as x CREATE(n)"""
-        writer = threading.Thread(target=thread_run_query, args=(graphs[0], heavy_write_query, 0))
-        writer.setDaemon(True)
-        writer.start()
+        writer = pool.apipe(thread_run_query, graphs[0], heavy_write_query, 0)
         redis_con.rename(new_graph, GRAPH_ID)
```

Comment:
Shouldn't this be: `redis_con.rename(GRAPH_ID, new_graph)`
See: https://redis-py.readthedocs.io/en/stable/index.html?highlight=rename#redis.Redis.rename